src/nvapi_gpu.cpp

```diff
@@ -310,11 +386,22 @@ extern "C" {
 
         auto architectureId = adapter->GetArchitectureId();
 
+        if (architectureId > NV_GPU_ARCHITECTURE_GA100) {
```

Comment:
Thinking about this, I would prefer to move those quirks further in the background. What about moving the `is*Needed` method to `env` and change the signature so that it looks like
```c++
if (needsAmpereSpoofing(architectureId, returnAddress))
    architectureId = NV_GPU_ARCHITECTURE_GA100;
```
(The check for Ada should move into that method.  I'm also fine to declare  `returnAddress` at the top of this method, even if it is unused for other architecture)

Once we have this, I can continue adding a quirk for https://github.com/jp7677/dxvk-nvapi/issues/47 (likely `needsPascalSpoofing(architectureId)` since we can just spoof a non DLSS architecture if the executable name matches that game.